app/pages/UcsfClientEligibilityPage/UcsfClientEligibilityPage.tsx

```diff
@@ -66,30 +40,63 @@ const ClientEligibilities = () => {
   };
 
   const toggleChecked = (eligibilityGroup: clientEligibilityGroup, eligibilityIndex: number) => {
-    const updatedEligibility = {
-      ...eligibilityGroup.eligibilities[eligibilityIndex],
-      checked: !eligibilityGroup.eligibilities[eligibilityIndex].checked,
-    };
-
-    const updatedEligibilities = [
-      ...eligibilityGroup.eligibilities.slice(0, eligibilityIndex),
-      updatedEligibility,
-      ...eligibilityGroup.eligibilities.slice(eligibilityIndex + 1),
-    ];
+    const targetEligibility = eligibilityGroup.eligibilities[eligibilityIndex];
+    const targetToggleState = !targetEligibility.checked;
+    let updatedEligibilities;
+
+    // If a user toggles "See all", we (un)check all of the eligibilities accordingly
+    if (targetEligibility.name === 'See all') {
+      updatedEligibilities = massToggleGroupEligibilities(
+        eligibilityGroup.eligibilities,
+        targetToggleState,
+      );
+    } else {
+      // If a user untoggles a specific eligiblity, this code also untoggles "See all"
+      // Currently, there is no logic to toggle "See all" if a user toggles all individual
+      // eligibilities given that this is somewhat of an implicit understanding; however, this
+      // can be added in the future
+      const seeAllEligibility = eligibilityGroup.eligibilities[0];
+      if (!targetToggleState) {
+        seeAllEligibility.checked = false;
+      }
+
+      const updatedEligibility = {
+        ...targetEligibility,
+        checked: !eligibilityGroup.eligibilities[eligibilityIndex].checked,
+      };
+
+      updatedEligibilities = [
+        { ...seeAllEligibility },
+        ...eligibilityGroup.eligibilities.slice(1, eligibilityIndex),
+        updatedEligibility,
+        ...eligibilityGroup.eligibilities.slice(eligibilityIndex + 1),
+      ];
+    }
 
     return {
       ...eligibilityGroup,
       eligibilities: updatedEligibilities,
     };
   };
 
+  // Toggles all eligibilities in accordance with the toggleState argument
+  const massToggleGroupEligibilities = (
+    eligibilities: clientEligibility[],
+    toggleState: boolean,
+  ) => (
+    eligibilities.map((eligibility: clientEligibility) => ({
+      ...eligibility,
+      checked: toggleState,
+    }))
+  );
+
   return (
     <div className={styles.eligibilitiesBox}>
       <div className={styles.eligibilitiesBox_title}>Client Identity</div>
       <ol className={styles.eligibilitiesLabels}>
 
         {/* Todo: This list rendering logic will be refactored when the API is setup */}
-        {eligibilityGroupList.map((eligibilityGroup, index) => (
+        {eligibilityGroupList.map((eligibilityGroup: clientEligibilityGroup, index: number) => (
```

Comment:
It's been a little while since I've written some significant TypeScript, but I feel like you should be able to elide these type annotations, since they should be inferable. Maybe the issue is that the initial declaration of `eligibilityGroupList` from the `useState()` call doesn't have enough info to allow TypeScript to infer its type?

Reply:
Creating a more specific `UcsfEligibilityDictionary` interface, as suggested below, fixed this. Thanks!